server/http_handler.go

```diff
@@ -647,6 +653,115 @@ func (h ddlHistoryJobHandler) ServeHTTP(w http.ResponseWriter, req *http.Request
 	return
 }
 
+func (h tableHandler) getPDAddr() ([]string, error) {
+	var pdAddrs []string
+	etcd, ok := h.store.(domain.EtcdBackend)
+	if !ok {
+		return nil, errors.New("not implemented")
+	}
+	pdAddrs = etcd.EtcdAddrs()
+	if len(pdAddrs) < 0 {
+		return nil, errors.New("pd unavailable")
+	}
+	return pdAddrs, nil
+}
+
+func (h tableHandler) addScatterSchedule(startKey, endKey []byte, name string) error {
+	pdAddrs, err := h.getPDAddr()
+	if err != nil {
+		return err
+	}
+	input := map[string]string{
+		"name":       "scatter-range-leader",
+		"start_key":  url.QueryEscape(string(startKey)),
+		"end_key":    url.QueryEscape(string(endKey)),
+		"range_name": name,
+	}
+	v, err := json.Marshal(input)
+	if err != nil {
+		return err
+	}
+	scheduleURL := fmt.Sprintf("http://%s/pd/api/v1/schedulers", pdAddrs[0])
+	resp, err := http.Post(scheduleURL, "application/json", bytes.NewBuffer(v))
+	if err != nil {
+		return err
+	}
+	if err := resp.Body.Close(); err != nil {
+		log.Error(err)
+	}
+	return nil
+}
+
+func (h tableHandler) deleteScatterSchedule(name string) error {
+	pdAddrs, err := h.getPDAddr()
+	if err != nil {
+		return err
+	}
+	scheduleURL := fmt.Sprintf("http://%s/pd/api/v1/schedulers/scatter-range-leader-%s", pdAddrs[0], name)
```

Comment:
IMHO, this API design is bad
Prefer Post to Get, and you can use simpler http client API